feat(cfn-include): add support for nested stacks (#8980)
aws/aws-cdk: packages/@aws-cdk/cloudformation-include/README.md

output.value = cfnBucket.attrArn;
```

## Nested Stacks

This module also support templates that use [nested stacks](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-nested-stacks.html).

For example, if you have the following parent template:

```json
{
  "Resources": {
    "ChildStack": {
      "Type": "AWS::CloudFormation::Stack",
      "Properties": {
        "TemplateURL": "https://my-s3-template-source.s3.amazonaws.com/child-import-stack.json",
        "Parameters": {
          "MyBucketParameter": "my-bucket-name"
        }
      }
    }
  }
}
```

where the child template pointed to by `https://my-s3-template-source.s3.amazonaws.com/child-import-stack.json` is:

```json
{
  "Parameters": {
    "MyBucketParameter": {
      "Type": "String",
      "Default": "default-bucket-param-name"
    }
  },
  "Resources": {
    "BucketImport": {
      "Type": "AWS::S3::Bucket",
      "Properties": {
        "BucketName": {
          "Ref": "MyBucketParameter"
        }
      }
    }
  }
}
```

> You can access the nested stack and perform all operations you would on any other CDK Stack if you include the parent stack as follows:

Reviewer: This phrasing reads a little backwards for me.... how about:

```suggestion
You can include both the parent stack and the nested stack in your CDK application as follows:
```

```typescript
const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
  templateFile: 'path/to/my-parent-template.json',
  nestedStacks: {
    'ChildStack': {
      templateFile: 'path/to/my-nested-template.json',
    },
  },
});
```

Now you can access the ChildStack nested stack and included template with:

```typescript
const includedChildStack = parentTemplate.getNestedStack('ChildStack');
const childStack: core.NestedStack = includedChildStack.stack;
const childStackTemplate: cfn_inc.CfnInclude = includedChildStack.includedTemplate;
```

> we can modify childStack:

Reviewer: ```suggestion
Now you can reference resources from `ChildStack` and modify them like any other included template:
```
